vm/src/pyobject.rs

```diff
@@ -948,44 +938,35 @@ pub trait DictProtocol {
     fn del_item(&self, key: &str);
 }
 
-impl DictProtocol for PyObjectRef {
-    fn contains_key(&self, k: &str) -> bool {
-        if let Some(dict) = self.payload::<PyDict>() {
-            objdict::content_contains_key_str(&dict.entries.borrow(), k)
-        } else {
-            unimplemented!()
-        }
-    }
-
-    fn get_item(&self, k: &str) -> Option<PyObjectRef> {
-        if let Some(dict) = self.payload::<PyDict>() {
-            objdict::content_get_key_str(&dict.entries.borrow(), k)
-        } else {
-            panic!("TODO {:?}", k)
-        }
-    }
+pub trait ItemProtocol {
+    // // Move: doesn't really belong in this protocol.
+    // fn get_key_value_pairs(&self, vm: &VirtualMachine) -> PyResult<PyListIterator>;
+    fn get_item<T: IntoPyObject>(&self, key: T, vm: &VirtualMachine) -> PyResult<PyObjectRef>;
```

Comment:
Can we use `PyResult` instead of `PyResult<PyObjectRef>`? I believe `PyObjectRef` is the default template parameter.